### PR TITLE
Backport: 18.0.2 backports

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: doc/release-notes.md [hunk 1]
```diff
@@ -54,246 +51,28 @@ is not recommended.
...
--------------
-A lot of refactoring, code cleanups and other small fixes were done in this release.
-
-v18.0.1 Change log
-==================
-
-See detailed [set of changes](https://github.com/dashpay/dash/compare/v0.17.0.3...dashpay:v18.0.1).
+This release resolves some excessive memory usage via the "evo" database (evodb).
+
+0.17.0.3 Change log
```
Comment (thelazier) [inline doc/release-notes.md:56]: should this be v18.0.2 Change log ?

File: doc/release-notes.md [hunk 2]
```diff
@@ -318,7 +97,7 @@ Dash Core tree 0.12.1.x was a fork of Bitcoin Core tree 0.12.
 
 These release are considered obsolete. Old release notes can be found here:
 
-- [v0.17.0.3](https://github.com/dashpay/dash/blob/master/doc/release-notes/dash/release-notes-0.17.0.3.md) released June/07/2021
```
Comment (thelazier) [inline doc/release-notes.md:321]: should this v0.17.0.3 line be not removed to be history ?

Comment (ogabrielides) [reply doc/release-notes.md:321]: @thelazier I believe here we only keep the last version's release notes, right @PastaPastaPasta ?

File: doc/release-notes.md [hunk 3]
```diff
@@ -54,246 +51,28 @@ is not recommended.
...
-A lot of refactoring, code cleanups and other small fixes were done in this release.
-
-v18.0.1 Change log
-==================
-
-See detailed [set of changes](https://github.com/dashpay/dash/compare/v0.17.0.3...dashpay:v18.0.1).
+This release resolves some excessive memory usage via the "evo" database (evodb).
+
+0.17.0.3 Change log
+===================
```
Comment (ogabrielides) [inline doc/release-notes.md:57]: @PastaPastaPasta Please correct version here

**Review by ogabrielides — CHANGES_REQUESTED** [review]

File: doc/release-notes.md (hunk 2 above)
Comment (UdjinM6) [reply doc/release-notes.md:321]: All previous releases should be listed here

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
pls add https://github.com/dashpay/dash/pull/5004/commits/3e2c2472b4d981b2936268549cee2283f310cb2e

**Review by ogabrielides — APPROVED** [review]
LGTM

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**